src/contracts/aludel/AludelV3Lib.sol

```diff
@@ -244,4 +244,25 @@ library AludelV3Lib {
         vault.totalStake += amount;
         aludel.totalStake += amount;
     }
+
+    function cleanupExpiredRewardSchedules(
+        IAludelV3.AludelData storage aludel
+    ) internal {
+        uint256 length = aludel.rewardSchedules.length;
+        for (uint index = 0; index < length; index++) {
+            IAludelV3.RewardSchedule memory schedule = aludel.rewardSchedules[index];
+
+            // we manage the length of the array manually to avoid accessing the array's length multiple times
+            length--;
```

Comment:
you're reducing the length *unconditionally* yet conditionally removing elements. This'll cause the function to iterate over only the first half of the array.

Please add a testcase triggering this error (a scenario where a RewardSchedule should be removed but isn't) and then fix it.